content/src/controller.js

```diff
@@ -786,6 +793,87 @@ function saveAction(forceOverwrite, loginPrompt, doneCallback) {
   });
 }
 
+function generateThumbnailDataUrl() {
+  var THUMBNAIL_SIZE = 128;
+  // Use the same background as the default icons.
+  var BACKGROUND_COLOR = '#eeeeee';
+
+  // Get the canvas inside the iframe.
+  var iframe = document.getElementsByTagName('iframe')[0];
```

Comment:
We need to be more specific about which iframe we are thumbnailing; gebtn('iframe')[0] means we can never add another iframe to the UI.   Pass the iframe as an argument to this function, and in the caller, locate it by either looking for ('.preview') or getPaneAtPosition('right').